dvc/repo/checkout.py

```diff
@@ -23,7 +23,9 @@ def get_all_files_numbers(stages):
     return sum(stage.get_all_files_number() for stage in stages)
 
 
-def checkout(self, target=None, with_deps=False, force=False, recursive=False):
+def _checkout(
+    self, target=None, with_deps=False, force=False, recursive=False
+):
```

Comment:
If you'll make `@ctx` reentrable this won't be needed, will also simplify things overall and make them more bullet-proof.